robocin/utility/fuzzy_compare.cpp

```diff
@@ -0,0 +1,41 @@
+//
+// Created by José Cruz <joseviccruz> on 08/04/23.
+//
+
+#include "robocin/utility/fuzzy_compare.h"
+
+namespace robocin {
+
+template class FuzzyIsZero<float>;
+template class FuzzyIsZero<double>;
+template class FuzzyIsZero<long double>;
+
+template class FuzzyEqualTo<float>;
+template class FuzzyEqualTo<double>;
+template class FuzzyEqualTo<long double>;
+
+template class FuzzyNotEqualTo<float>;
+template class FuzzyNotEqualTo<double>;
+template class FuzzyNotEqualTo<long double>;
+
+template class FuzzyThreeWay<float>;
+template class FuzzyThreeWay<double>;
+template class FuzzyThreeWay<long double>;
+
+template class FuzzyLess<float>;
+template class FuzzyLess<double>;
+template class FuzzyLess<long double>;
+
+template class FuzzyLessEqual<float>;
+template class FuzzyLessEqual<double>;
+template class FuzzyLessEqual<long double>;
+
+template class FuzzyGreater<float>;
+template class FuzzyGreater<double>;
+template class FuzzyGreater<long double>;
+
+template class FuzzyGreaterEqual<float>;
```

Comment:
Pegando essa função por exemplo, acho uma boa discussão quantas palavras é aceitável comer, principalmente em funções mais públicas. Por ex, o nome completo dessa função é: `FuzzyIsGreaterOrEqual`. Isso permite que só lendo o nome da função se saiba o que retorna, e mesmo parecendo obvio que é Or, acho legal botar.
Como disse pode ser um olhar enviesado de quem coda muito em Java.

Reply:
Concordo com @bcs5, mas acho que temos que levar em consideração que se for feita com base em algo que já existe na documentação padrão, mudar muito talvez significasse aumentar a carga cognitiva de quem for usar por fugir do que já é conhecido. Não digo isso para não mudar, acho que seria mais intuitivo e auto-documentável (talvez também enviesada por Java), mas só para levar em consideração esse fator. O mesmo se aplicaria nos comentários relacionados a booleano com isEqual etc...

Reply:
Para essa biblioteca, assimilar-se com o padrão parece o desejado, e foi o que pesou para a escolha dos nomes, como comentado na [conversation](https://github.com/robocin/robocin-cpp/pull/1/files#r1228730324) acima.

Mas concordo com ambos, apesar do padrão em C++ possuir nomes mais curtos, as vezes até mesmo a união de palavras (e.g. toupper, tolower), a bibiloteca - quando não houver uma proximidade em termos de implementação com classes já existentes -, irá levar em consideração um entendimento mais claro daquilo que se propõe a fazer.